### PR TITLE
single pool wallet for every job created

### PR DESCRIPTION
**Pull request by dhruvja**

### CODE REVIEW

**Comment by dhruvja** [comment]
there are some changes which are left, will add soon

File: programs/job/src/lib.rs [hunk 1]
```diff
@@ -41,15 +99,48 @@ pub struct Initialize<'info> {
...
+    pub job_account: Box<Account<'info, JobStakingParameter>>,
+    pub token_mint: Account<'info, Mint>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(mut)]
+    pub escrow_wallet_state: Account<'info, TokenAccount>,
+    #[account(
+        mut,
+        // constraint=wallet_to_withdraw_from.owner == authority.key(),
+        // constraint=wallet_to_withdraw_from.mint == token_mint.key()
```
Comment (blasrodri) [inline programs/job/src/lib.rs:115]:
```suggestion
```

File: programs/job/src/lib.rs [hunk 2]
```diff
@@ -26,6 +29,61 @@ pub mod job {
...
+            let cpi_ctx = CpiContext::new_with_signer(
+                ctx.accounts.token_program.to_account_info(),
+                transfer_instruction,
+                outer.as_slice(), //signer PDA
+            );
+    
+            let amount_in_64 = amount as u64;
+    
+            // The `?` at the end will cause the function to return early in case of an error.
+            // This pattern is common in Rust.
```
Comment (blasrodri) [inline programs/job/src/lib.rs:80]:
```suggestion
```

File: programs/job/src/lib.rs [hunk 3]
```diff
@@ -26,6 +29,61 @@ pub mod job {
...
+
+        let candidate_staking_program_id: &str = "BF1jhf5eA5X1Tu8JByv8htnkUaG6WzmYEMLx2kbZ7YiW";
+
+        let ixns = ctx.accounts.instructions.to_account_info();
+        let current_index = tx_instructions::load_current_index_checked(&ixns)? as usize;
+        let current_ixn = tx_instructions::load_instruction_at_checked(current_index, &ixns)?;
+
+        msg!("Current program ID: {} job program ID: {}", current_ixn.program_id, *ctx.program_id);
+
+        // let candidate_parameters = &mut ctx.accounts.candidate_account;
```
Comment (blasrodri) [inline programs/job/src/lib.rs:50]:
```suggestion
```

File: programs/candidate_staking/src/lib.rs [hunk 4]
```diff
@@ -210,39 +211,43 @@ pub mod candidate_staking {
...
     pub token_program: Program<'info, Token>,
     pub rent: Sysvar<'info, Rent>,
 }
+
 #[derive(Accounts)]
 #[instruction(job_ad_id: String, application_id: String, base_bump: u8, general_bump: u8, application_bump: u8, job_bump: u8, wallet_bump: u8)]
 pub struct Stake<'info> {
     #[account(mut, seeds = [CANDIDATE_SEED, application_id.as_bytes()[..18].as_ref(), application_id.as_bytes()[18..].as_ref() ,authority.key().as_ref()],bump = base_bump)]
-    pub base_account: Account<'info, CandidateParameter>,
+    pub base_account: Box<Account<'info, CandidateParameter>>,
```
Comment (blasrodri) [inline programs/candidate_staking/src/lib.rs:241]: why box? what happens if we don't?

File: programs/candidate_staking/Cargo.toml [hunk 5]
```diff
@@ -19,7 +19,7 @@ default = []
 overflow-checks = true
 
 [dependencies]
-anchor-lang = "0.24.2"
+anchor-lang = {version = "0.24.2", features = ["init-if-needed"]}
```
Comment (blasrodri) [inline programs/candidate_staking/Cargo.toml:22]: can you please elaborate on why we need this?

**Comment by blasrodri** [comment]
Check https://github.com/coral-xyz/anchor/issues/1191#issuecomment-1005188435 for `init-in-needed`. We should try to avoid

**Review by blasrodri — CHANGES_REQUESTED** [review]
Can we try to do this without init-if-needed?

**Comment by dhruvja** [comment]
> Can we try to do this without init-if-needed?

We can but since it's a token account which is getting initialized, I don't see any problem in using init-if-needed.

**Comment by blasrodri** [comment]
> > Can we try to do this without init-if-needed?
> 
> We can but since it's a token account which is getting initialized, I don't see any problem in using init-if-needed.

Can you explain me why we need this feature?

**Pull request closed without merging**